server/params/index.js

```diff
@@ -34,17 +34,31 @@ module.exports = (app) => {
      * Userid.
      */
     userid: (req, res, next, userid) => {
-      parseId(userid)
-        .then(id => User.findById(id))
-        .then((user) => {
-          if (!user) {
-            return next(new errors.NotFound(`User '${userid}' not found`));
-          } else {
-            req.user = user;
-            next();
-          }
-        })
-        .catch(next)
+
+      const callback = (user) => {
```

Comment:
Or to avoid callback and duplication with `groupid` code:

```
  function getByIdOrName(model, idOrName, nameKey) {
    if (isNaN(idOrName)) { // name
      return model.find({
        where: { [nameKey]: idOrName.toLowerCase() }
      });
    } else { // id
      return model.findById(idOrName);
    }
  }

  /**
   * Public methods.
   */
  return {

    /**
     * UserId or username.
     */
    userid: (req, res, next, idOrName) => {
      getByIdOrName(User, idOrName, "username").tap(user => {
        if (!user) {
          throw new errors.NotFound(`User '${idOrName}' not found`);
        }
      })
      .then(user => req.user = user)
      .asCallback(next);
    },

    /**
     * GroupId or slug.
     */
    groupid: (req, res, next, groupIdOrName) => {
      getByIdOrName(Group, groupIdOrName, "slug").tap(group => {
        if (!group) {
          return next(new errors.NotFound(`Group '${groupid}' not found`));
        }
      })
      .then(group => req.group = group)
      .asCallback(next);
    },
    ...
  }
```